### DISCUSSION TITLE
Compiling OpenSSL 3.0.x with Rhel7 and gcc 4.8.2

### DISCUSSION THREAD

**Discussion by MohitS05** [openssl/openssl#24071] [Q&A]
I am trying to compile OpenSSL 3.0.x for an old system. For support reasons, we can't update it. The specs of system are as:
Rhel7, Gcc 4.8.2 
Are these configurations supported on 3.0.x ? 
I am unable to update Perl on the machine, and keep getting error on running Configure or when trying to update perl.   
Please suggest.

**Comment by tom-cosgrove-arm** [comment]
It would be useful if you included the version of Perl, and the error messages you are seeing - then someone might be able to give you a workaround

**Reply by MohitS05** [reply]
Error is -
Can't locate IPC/Cmd.pm in @INC (@INC contains: /usr/local/karol/openssl-3.0.13/util/perl /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 . /usr/local/karol/openssl-3.0.13/external/perl/Text-Template-1.56/lib) at /usr/local/karol/openssl-3.0.13/util/perl/OpenSSL/config.pm line 19.
BEGIN failed--compilation aborted at /usr/local/karol/openssl-3.0.13/util/perl/OpenSSL/config.pm line 19.
Compilation failed in require at ./Configure line 23.
BEGIN failed--compilation aborted at ./Configure line 23.

Perl Version:
5.16.3 perl

**Reply by t8m** [reply]
You need to install IPC/Cmd.pm. Use: `yum install 'perl(IPC::Cmd)'`

**Reply by MohitS05** [reply]
This gives an error:
This fails after saying -
```
Error downloading packages:
perl-Module-Metadata-1.000018-2.el7.noarch: [Errno 256] No more mirrors to try.
gdbm-devel-1.10-8.el7.x86_64: [Errno 256] No more mirrors to try.
1:perl-IPC-Cmd-0.80-4.el7.noarch: [Errno 256] No more mirrors to try.
1:perl-Params-Check-0.38-2.el7.noarch: [Errno 256] No more mirrors to try.
1:perl-Module-Load-0.24-3.el7.noarch: [Errno 256] No more mirrors to try.
perl-ExtUtils-MakeMaker-6.68-3.el7.noarch: [Errno 256] No more mirrors to try.
pyparsing-1.5.6-9.el7.noarch: [Errno 256] No more mirrors to try.
perl-Locale-Maketext-1.23-3.el7.noarch: [Errno 256] No more mirrors to try.
perl-ExtUtils-Manifest-1.61-244.el7.noarch: [Errno 256] No more mirrors to try.
```

**Reply by t8m** [reply]
You can try to find these packages manually from CentOS 7 or some other RHEL-7 clone.

**Reply by MohitS05** [reply]
Hi, 
We couldn't find any mirrors for those versions. We tried with closest packages, but still no luck. It ended up in a seemingly endless list of dependencies  requiring installation.

**Reply by t8m** [reply]
Unfortunately there is not much more we can do here to help. These packages are available in RHEL-7 but you of course need proper support license for RHEL to be able to register to the RHN and download them. You could try to install CentOS 7, build the OpenSSL there and transfer the compiled binaries to RHEL-7. They should be compatible.
